sklearn/datasets/_samples_generator.py

```diff
@@ -1294,6 +1310,19 @@ def make_sparse_coded_signal(
     # encode signal
     Y = np.dot(D, X)
 
+    # raise warning if data_transposed is not passed explicitly
+    if data_transposed == "deprecated":
```

Comment:
In the current state of this PR, we are not deprecating the parameter right away. Instead we are warning that the default is going to change:

```suggestion
    if data_transposed == "warn":
```

This is inline with the [Change the default value of a parameter](https://scikit-learn.org/stable/developers/contributing.html#change-the-default-value-of-a-parameter) docs.

Then in 1.3, I assume we will deprecate the parameter.